File: server/src/WebAPI.Contracts/Article/CreateArticleRequest.cs
namespace WebAPI.Contracts.Article;

public sealed record CreateArticleRequest(
    string ArticleId,
    string Title,
    string Description,
    string Text,
    bool IsPublished,
    bool IsPinned
)
{
    ///<example>some-article-id</example>
    public string ArticleId { get; init; } = ArticleId;
    ///<example>Заголовок статьи</example>
    public string Title { get; init; } = Title;
    ///<example>Описание статьи</example>
    public string Description { get; init; } = Description;
    ///<example>Содержимое статьи</example>
    public string Text { get; init; } = Text;
    ///<example>true</example>
    public bool IsPublished { get; init; } = IsPublished;
    ///<example>false</example>
    public bool IsPinned { get; init; } = IsPinned;
}

File: new-client/src/entities/article/model/createArticleRequest.ts
import { z } from "zod"

export const createArticleRequest = z.object({
  articleId: z.string()
    .min(3, { message: `Строка должна содержать минимум 3 символа` })
    .max(64, { message: `Строка должна содержать максимум 64 символа` })
    .regex(/^[a-zA-Z0-9-]+$/),
  title: z.string()
    .min(3, { message: `Строка должна содержать минимум 3 символа` })
    .max(64, { message: `Строка должна содержать максимум 64 символа` }),
  description: z.string().max(64, { message: `Строка должна содержать максимум 64 символа` }),
  text: z.string(),
  isPublished: z.boolean(),
  isPinned: z.boolean()
})
